src/app/clusters/closure-dimension-server/closure-dimension-cluster-logic.cpp

```
@@ -0,0 +1,500 @@
+    {
+        if (mConformance.HasFeature(Feature::kPositioning))
+        {
+            VerifyOrReturnError(target.position.Value() <= PERCENT100THS_MAX_VALUE, CHIP_ERROR_INVALID_ARGUMENT);
+        }
+        else
+        {
+            ChipLogDetail(Zcl, "Disregarding the position value in the target since the Positioning feature is not enabled.");
+            target.position.ClearValue();
+        }
+    }
+
+    if (target.latch.HasValue())
+    {
+        if (mConformance.HasFeature(Feature::kMotionLatching))
+        {
+            VerifyOrReturnError(EnsureKnownEnumValue(target.latch.Value()) != TargetLatchEnum::kUnknownEnumValue,
+                                CHIP_ERROR_INVALID_ARGUMENT);
+        }
+        else
+        {
+            ChipLogDetail(Zcl, "Disregarding the latch value in the target since the MotionLatching feature is not enabled.");
+            target.latch.ClearValue();
+        }
+    }
+
+    if (target.speed.HasValue())
+    {
+        if (mConformance.HasFeature(Feature::kSpeed))
+        {
+            VerifyOrReturnError(EnsureKnownEnumValue(target.speed.Value()) != Globals::ThreeLevelAutoEnum::kUnknownEnumValue,
+                                CHIP_ERROR_INVALID_ARGUMENT);
+        }
+        else
+        {
+            ChipLogDetail(Zcl, "Disregarding the speed value in the target since the Speed feature is not enabled.");
+            target.speed.ClearValue();
+        }
+    }
+
+    // TODO: Target.Position value SHALL follow the scaling from "Resolution Attribute".
+    if (target != mState.target)
+    {
+        mState.target = target;
+        mMatterContext.MarkDirty(Attributes::Target::Id);
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR ClusterLogic::SetResolution(const Percent100ths resolution)
+{
+    VerifyOrReturnError(mInitialized, CHIP_ERROR_INCORRECT_STATE);
+
+    VerifyOrReturnError(mConformance.HasFeature(Feature::kPositioning), CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE);
```

So why does this return an error on feature mismatch while
 other setters just silently ignore invalid values?  And why is none of this documented?

---

refactoring code to return CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE, if feild is provided without the feature support